### PR TITLE
add class to generate configs

### PR DESCRIPTION
**Pull request by trefzer**
The new class openssl::configs create openssl.conf files.
The generation bases on a new .epp template since epp can
use parameters which was easier to integrate.

closes: #41

### CODE REVIEW

File: manifests/configs.pp [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  Optional[Array]   $extendedkeyusages = undef,
+  Optional[Array]   $keyusages         = undef,
+  Optional[Array]   $subjectaltnames   = undef,
+  Hash              $conffiles         = {},
+){
+
+  # dpendencies: ensure config file is generated before potential usage
+  File<| tag=='openssl-configs' |> -> Ssl_pkey<| |>
+  File<| tag=='openssl-configs' |> -> X509_cert<| |>
+  File<| tag=='openssl-configs' |> -> X509_request<| |>
```
Comment (raphink) [inline manifests/configs.pp:75]: I believe https://github.com/camptocamp/puppet-openssl/pull/117 makes this moot, as each type now autorequires its config file.

File: manifests/configs.pp [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+  $conffiles.each | String $filename, Hash $vals| {
```
Comment (raphink) [inline manifests/configs.pp:77]: Nitpick
```suggestion
  $conffiles.each |String $filename, Hash $vals| {
```

File: manifests/configs.pp [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    file { $filename:
+      ensure  => pick($vals['ensure'],'present'),
+      owner   => pick($vals['owner'],$owner),
+      group   => pick($vals['group'],$group),
+      mode    => pick($vals['mode'],$mode),
```
Comment (raphink) [inline manifests/configs.pp:82]: Style nitpick

```suggestion
      ensure  => pick($vals['ensure'], 'present'),
      owner   => pick($vals['owner'], $owner),
      group   => pick($vals['group'], $group),
      mode    => pick($vals['mode'], $mode),
```

File: manifests/configs.pp [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+        unit              => $unit,
+        email             => $email,
+        default_bits      => $default_bits,
+        default_md        => $default_md,
+        default_keyfile   => $default_keyfile,
+        basicconstraints  => $basicconstraints,
+        extendedkeyusages => $extendedkeyusages,
+        keyusages         => $keyusages,
+        subjectaltnames   => $subjectaltnames,
+      } + delete($vals,['ensure','owner','group','mode']) ),
```
Comment (raphink) [inline manifests/configs.pp:97]:
```suggestion
      } + delete($vals,['ensure', 'owner', 'group', 'mode'])
    ),
```

File: manifests/configs.pp (hunk 1 above)
Comment (trefzer) [reply manifests/configs.pp:75]: ack

File: manifests/configs.pp (hunk 2 above)
Comment (trefzer) [reply manifests/configs.pp:77]: ack

**Pull request merged**